_posts/2016-04-04-code-smells-part-II.md

```diff
@@ -0,0 +1,91 @@
+So while talking to [David](https://twitter.com/DHatanian), he asked me: and what happens if I encapsulate the switch into a method, is it acceptable then? That's really a good question... If your switch case is only used to "take care" of one behaviour and that's it, then it might be ok . Remember identifying a code smell doesn't mean that you have to get always ride of it: it's a trade of. If you find your switch statement replicated and each replication has  different behaviour, then you cannot simply isolate the switch statement in a method. You need to find a proper "home" for it to be in. As a rule of thumb, you should think of polymorphism when you find yourself in this situation. There are two refactoring techniques that we can apply here: 
+
+ - **_Replace Type Code with Subclasses_**
+   This technique consists of creating subclasses for each switch case and applying the respective behaviour to these subclasses.
+ - **_Replace Type Code With Strategy_**
+   Similar to the above one, in this case, you should make use of one of the patterns: [State](https://en.wikipedia.org/wiki/State_pattern) or [Strategy](https://en.wikipedia.org/wiki/Strategy_pattern).
+
+So when to use one or the other? If the **_Type Code_** does not change the behaviour of a class you can use the **_Subclasses_** technique. Separating each behaviour in it's appropriated subclass will enforce the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) and make the code more readable in general. If you need to add another case, you just add a new class to your code without having to modify any other code. So you apply the [Open/Close Principle](https://en.wikipedia.org/wiki/Open/closed_principle).
+You should use the Strategy approach when the **_Type Code_** affects the behaviour of your classes. If you're changing the state of the class, fields and many other actions then you should use the [State Pattern](https://en.wikipedia.org/wiki/State_pattern). if it only affects how you select a behaviour of the class then the [Strategy Pattern](https://en.wikipedia.org/wiki/Strategy_pattern) is your man.
+
+#### Temporary Field
+
+This case occurs when we are calculating some big algorithm that needs several input variables. Creating these fields in the class has no value most of the times because they are just used for this specific calculation. And this can be dangerous too because you have to be sure you reinitialize them before you start the next computation. 
+Here the best refactoring technique is to use **_Replace Method with Method Object_**, which will extract the method into a separate class. Then you can split the method into several methods within the same class. 
+
+
+#### Refused Bequest
+
+This code smell is a little tricky to detect because this happens when a subclass doesn't use all the behaviour of its parent class. So it's as if the subclass "refuses" some behaviour("bequest") of its parent class. 
```

Comment:
Is this about `refusing` or `reusing`? Could be made clearer with an example, too.